Weather/ViewModel/DailyForecastViewModel.swift

```diff
@@ -23,16 +23,19 @@ class DailyForecastViewModel: ViewModel {
     }
   }
   
-  weak var view: DailyForecastController?
-
-	lazy var dataSet = {
-		ObservableList<DailyForecast>()
-	}()
-	
-	lazy var dataSource = {
-		DailyForecastDataSource(dataSet: dataSet)
-	}()
-	
+  weak var view: DailyForecastController? {
+    didSet {
+      if let view = view {
```

Comment:
Bu gibi durumlarda view yoksa scope da başka hiçbir işlem olmayacağı için guard yazımı daha doğru görüntü açısından. baya kullanan var sen sevmiyorsun biliyorum :D 

guard let view = view else { return }
.....